drivers/i2c/Kconfig

```diff
@@ -54,3 +50,7 @@ config I2CMULTIPLEXER_PCA9540BDP
 endmenu # I2C Multiplexer Support
 
 endif
+
+config ARCH_HAVE_I2CRESET
```

Comment:
We have been through this one before.  ARCH_HAVE_I2CRESET must be outside of if I2C, otherwise Kconfig generates errors.  This cannot be accepted into the repository.

That same mistake was made in a previous PR and that had to be reverted because of all of the configuration errors it causes.  This cannot be accepted into the repository.

Reply:
The reason is that  ARCH_HAVE_I2CRESET  is selected unconditionally in all architecture configurations that support I2C reset.  That is correct, advertising that an architecture supports a feature is conditional.  CONFIG_I2C is completely unnecessary to enable architecture I2C support.

If conditions are added to ARCH_HAVE_I2CRESET, the errors referring to unmet dependencies will be generated on all configurations that enabled architecture-specific I2C support but don't need CONFIG_I2C.

This same change was added a few weeks ago and this resulted in several new errors in build testing and had to be fixed by moving ARCH_HAVE_I2CRESET  outside of I2C.  It must be that way.

Reply:
You could move the ARCH_HAVE_I2CRESET  item into the drivers/Kconfig file before CONFIG_I2C is selected.  That would resolve the problem.

I personally believe that all of the ARCH_HAVE_ definitions belong in the arch/Kconfig menu.  That is were they were originally and there is where they best belong.  The ARCH naming conventions is only correct if they reside in the arch/Kconfig file.  But there were moved all over in a recent PR and now the naming is wrong and a lot of dependencies relationships have been broken and kludged around like this one.

Reply:
This Kconfig file isn't conditionally brought in when I2C is defined. So technically ARCH_HAVE_I2CRESET is declared at the same level as it was before. If it would be preferred for these to be declared in the larger Kconfig as before, I can change that.

Reply:
i suppose it could even go at the end of drivers/i2c/Kconfig

That is not very comfortable.  But I am not comfortable with the breaking of the naming convention either.  The first word of the configuration name is suppose to be a clue where you can find the configuration definition.  So everything in drivers/Kconfig should be named DRIVERS_, everything under drivers/i2c/Kconfig should be named I2C.

This belong in arch/Kconfig.  Evertything in arch/Kconfig should be named ARCH_